core/services/offchainreporting/job.go

```diff
@@ -0,0 +1,149 @@
+package offchainreporting
+
+import (
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+	"github.com/shopspring/decimal"
+
+	"github.com/smartcontractkit/chainlink/core/services/job"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+	"github.com/smartcontractkit/chainlink/core/utils"
+	ocrcontract "github.com/smartcontractkit/offchain-reporting-design/contract"
+	"github.com/smartcontractkit/offchain-reporting-design/prototype/gethwrappers/ulairi"
+	ocr "github.com/smartcontractkit/offchain-reporting-design/prototype/offchainreporting"
+	ocrconfig "github.com/smartcontractkit/offchain-reporting-design/prototype/offchainreporting/config"
+)
+
+const JobType = "offchainreporting"
+
+type JobSpec struct {
+	ID                models.ID      `json:"id"`
+	ContractAddress   common.Address `json:"contractAddress"`
+	P2PNodeID         string         `json:"p2pNodeID"`
+	P2PBootstrapNodes []struct {
+		PeerID    string `json:"peerID"`
+		Multiaddr string `json:"multiAddr"`
+	} `json:"p2pBootstrapNodes"`
+	KeyBundle          string         `json:"keyBundle"`
+	MonitoringEndpoint string         `json:"monitoringEndpoint"`
+	NodeAddress        common.Address `json:"nodeAddress"`
+	ObservationTimeout time.Duration  `json:"observationTimeout"`
+	ObservationSource  job.Fetcher    `json:"observationSource"`
+}
+
+// JobSpec conforms to the job.JobSpec interface
+var _ job.JobSpec = JobSpec{}
+
+func (spec JobSpec) JobID() *models.ID {
+	return &spec.ID
+}
+
+func (spec JobSpec) JobType() string {
+	return JobType
+}
+
+func RegisterJobTypes(jobSpawner job.Spawner, orm ormInterface) {
+	jobSpawner.RegisterJobType(
+		JobType,
+		func(jobSpec job.JobSpec) (job.JobService, error) {
+			concreteSpec, ok := jobSpec.(JobSpec)
+			if !ok {
+				return nil, errors.Errorf("expected an offchainreporting.JobSpec, got %T", jobSpec)
+			}
+
+			db := database{JobSpecID: concreteSpec.ID, orm: orm}
+
+			config, err := db.ReadConfig()
+			if err != nil {
+				return nil, err
+			}
+
+			var backend bind.ContractBackend    // @@TODO
+			var netEndpoint ocr.NetworkEndpoint // @@TODO
+			return ocr.NewOracle(
+				&config,
+				netEndpoint,
+				dataSource(concreteSpec.ObservationSource),
+				ulairi.NewUlairi(concreteSpec.ContractAddress, backend),
+			), nil
+		},
+	)
+}
+
+// dataSource is a simple wrapper around an existing job.Fetcher that converts
```

Comment:
I believe it actually requires an `int64` in its current incarnation... if not, then it should because thats what gets reported onchain.

Reply:
On-chain we can get up to an int256, so I talked to @kaleofduty about switching that `int64` to a `*big.Int` for future-proofing if nothing else